src/sentry/integrations/gitlab/integration.py

```diff
@@ -332,7 +332,16 @@ def get_pipeline_views(self):
 
     def build_integration(self, state):
         data = state["identity"]["data"]
-        oauth_data = get_oauth_data(data)
+
+        # Gitlab requires the client_id and client_secret for refreshing the access tokens
+        client_id = state.get("oauth_config_information", {}).get("client_id")
+        client_secret = state.get("oauth_config_information", {}).get("client_secret")
+        oauth_data = {
+            **get_oauth_data(data),
+            "client_id": client_id,
+            "client_secret": client_secret,
```

Comment:
we will store `ouath_data` object on the Identity.data column

Reply:
Nit, I'd do:
```py
configs = state.get("oauth_config_information", {})
oauth_data = {
    **get_oauth_data(data),
    "client_id": configs.get("client_id"),
    "client_secret": configs.get("client_secret"),
}
```